spec/overcommit/hook/pre_commit/licence_header_spec.rb

```diff
@@ -0,0 +1,29 @@
+require 'spec_helper'
+
+describe Overcommit::Hook::PreCommit::LicenceHeader do
+  let(:config)  { Overcommit::ConfigurationLoader.default_configuration }
+  let(:context) { double('context') }
+  subject { described_class.new(config, context) }
+
+  context 'when licence file is missing' do
+    before do
+      subject.stub(:licence_lines).and_raise(Errno::ENOENT)
+    end
+
+    it { should fail_hook }
+  end
```

Comment:
Can rewrite this to be an "actual" test without stubs by writing (untested):

```ruby
around do |example|
  repo do
    example.run
  end
end 

context 'when license file is missing' do
  it { should fail_hook }
end

context 'when license file exists' do
  let(:license_contents) { <<-LICENSE }
  Look at me
  I'm a license
  LICENSE

  let(:file) { 'some-file.txt' }

  before do
    File.open('LICENSE.txt', 'w') { |f| f.write(license_contents) }
    subject.stub(:applicable_files).and_return([file])
  end

  context 'when all files contain the license header' do
    before do
      File.open(file, 'w') { |f| f.puts(license_contents); f.write('And some text') }
    end
 
    it { should pass }
  end

  context 'when all files contain the license header' do
    before do
      File.open(file, 'w') { |f| f.write('Some text without a license') }
    end

    it { should fail_hook }
  end
end
```

We use stubbing in the Overcommit test suite when integrating with third-party tools, since we don't want to require developers to install those tools locally in order to run tests for Overcommit. In this case, creating actual files is totally fine.